src/components/NavBar/Util.tsx

```diff
@@ -0,0 +1,172 @@
+import React from 'react';
+import {
+  Box,
+  Link,
+  Text,
+  Button,
+  Image,
+  Stack,
+  Flex,
+  Heading,
+} from '@chakra-ui/react';
+import {CloseIcon} from '@chakra-ui/icons';
+import {FaBars, FaGlobeAmericas} from 'react-icons/fa';
+import logo512 from './logo512.png';
+import {useTranslation} from 'react-i18next';
+
+/**
+ * Props for Menu Toggle Button
+ */
+interface MenuToggleProps {
+  toggle: () => void;
+  isOpen: boolean;
+}
+
+/**
+ * Menu toggle button for mobile-sized screens
+ * @param toggle - function that toggles whether the menu is open or closed
+ * @param isOpen - boolean, is the menu open
+ * @returns a button that toggles the menu when clicked
+ */
+const MenuToggle: ({toggle, isOpen}: MenuToggleProps) => JSX.Element = ({
+  toggle,
+  isOpen,
+}: MenuToggleProps) => {
+  return (
+    <Button
+      variant="ghost"
+      _hover={{background: 'teal'}}
+      onClick={toggle}
+      width="60px"
+    >
+      {isOpen ? <CloseIcon size="lg" /> : <FaBars size="lg" />}
+    </Button>
+  );
+};
+
+/**
+ * Props for the Language Toggle
+ */
+interface LanguageToggleProps {
+  toggle: () => void;
+}
+
+/**
+ * Button to change the language between English and Spanish
+ * @param toggle - function that toggles the website language
+ * @returns a button that toggles the language when clicked
+ */
+const LanguageToggle: ({toggle}: LanguageToggleProps) => JSX.Element = ({
+  toggle,
+}: LanguageToggleProps) => {
+  const {t} = useTranslation('menu');
+  return (
+    <Button
+      variant="ghost"
+      _hover={{background: 'teal', fontWeight: 'bold'}}
+      id="changeLanguage"
+      onClick={toggle}
+      mt={0}
+    >
+      <FaGlobeAmericas />
+      <Text paddingLeft={2} fontSize="xl">
+        {t('changeLanguage')}
+      </Text>
+    </Button>
+  );
+};
+
+/**
+ * Props for a single item in the navigation menu
+ */
+interface MenuItemProps {
+  label: string;
+  href: string;
+}
+
+/**
+ * A single item in the navigation menu
+ * @param label - user-facing text for the item
+ * @param href - the page that this item routes to
+ * @returns a navigation link
+ */
+const MenuItem: ({label, href}: MenuItemProps) => JSX.Element = ({
+  label,
+  href,
+}: MenuItemProps) => {
+  return (
+    <Link href={href} textAlign="center">
+      <Text _hover={{fontWeight: 'bold'}} fontSize="xl">
+        {label}
+      </Text>
+    </Link>
+  );
+};
+
+/**
+ * @returns the logo for the top left corner of the site
+ */
+const Logo: () => JSX.Element = () => {
+  return (
+    <Flex flexDir="row" alignItems="center">
+      <Image width="60px" src={logo512}></Image>
+      <Heading
+        display={['none', 'none', 'none', 'inline']}
```

Comment:
I think it'd be good to display this in general (not just when the browser is wider), and maybe remove the "South Gate Air Quality" that's at the top of the map page. I also think this would fit well on the navbar when it's mobile, since the only thing in the navbar for mobile is the logo and the bars